app/code/Magento/InventorySales/Plugin/Inventory/StockRepository/SalesChannelDataAfterGetPlugin.php

```diff
@@ -0,0 +1,52 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventorySales\Plugin\Inventory\StockRepository;
+
+use Magento\InventoryApi\Api\Data\StockExtension;
+use Magento\InventoryApi\Api\Data\StockInterface;
+use Magento\InventoryApi\Api\StockRepositoryInterface;
+use Magento\InventorySales\Model\GetSalesChannelsByStockInterface;
+
+class SalesChannelDataAfterGetPlugin
+{
+    /**
+     * @var GetSalesChannelsByStockInterface
+     */
+    private $getSalesChannelByStock;
+
+
+
+    /**
+     * SalesChannelDataAfterGetPlugin constructor.
+     * @param GetSalesChannelsByStockInterface $getSalesChannelByStock
+     */
+    public function __construct(
+        GetSalesChannelsByStockInterface $getSalesChannelByStock
+    ) {
+        $this->getSalesChannelByStock = $getSalesChannelByStock;
+    }
+
+    public function afterGet(StockRepositoryInterface $subject, StockInterface $result): StockInterface
+    {
+        /** @var StockExtension $extensionAttributes */
+        $extensionAttributes = $result->getExtensionAttributes();
+        $salesChannelSearchResults = $this->getSalesChannelByStock->get((int)$result->getStockId());
+        $extensionAttributes->setData('sales_channels', $salesChannelSearchResults);
```

Comment:
Do not use `setData` method, you could use method which was described `extension_attributes.xml`

Reply:
done